src/opera/parser/tosca/csar.py

```diff
@@ -0,0 +1,101 @@
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+
+import yaml
+from opera.error import ParseError
+
+
+class CloudServiceArchive:
+    def __init__(self, csar_name, csar_folder_path):
+        self._csar_name = csar_name
+        self._csar_folder_path = csar_folder_path
+        self._tosca_meta = None
+        self._root_yaml_template = None
+        self._metadata = None
+
+    def validate_csar(self):
+        with TemporaryDirectory(dir=self._csar_folder_path) as tempdir:
+            ZipFile(self._csar_name, 'r').extractall(tempdir)
+
+            try:
+                if Path(tempdir).joinpath("TOSCA-Metadata").joinpath("TOSCA.meta").exists():
+                    meta_file = open(Path(tempdir).joinpath("TOSCA-Metadata").joinpath("TOSCA.meta"), 'r')
+                    self._tosca_meta = yaml.safe_load(meta_file)
+                    meta_file.close()
+
+                    self._get_csar_version()
+                    self._get_tosca_meta_file_version()
+                    template_entry = self._get_entry_definitions()
+                    self._get_created_by()
+                    self._get_other_definitions()
+
+                    # check if 'Entry-Definitions' points to an existing template file in the CSAR
+                    if not Path(tempdir).joinpath(template_entry).exists():
+                        raise ParseError('The file "{}" defined within "Entry-Definitions" in '
+                                         '"TOSCA-Metadata/TOSCA.meta" does not exist.'.format(template_entry), self)
+
+                    return template_entry
+                else:
+                    root_yaml_files = []
+
+                    root_yaml_files.extend(Path(tempdir).glob('*.yaml'))
+                    root_yaml_files.extend(Path(tempdir).glob('*.yml'))
+
+                    if len(root_yaml_files) != 1:
+                        raise ParseError("There should be one root level yaml "
+                                         "file in the root of the CSAR: {}.".format(root_yaml_files), self)
+
+                    root_template = open(Path(root_yaml_files[0]), "r")
+                    root_yaml_template = yaml.safe_load(root_template)
+                    self._metadata = root_yaml_template.get('metadata')
+                    root_template.close()
+
+                    self._get_template_version()
+                    self._get_template_name()
+                    self._get_author()
+
+                    return root_yaml_files[0].name
+            except Exception as e:
+                raise ParseError("Invalid CSAR structure: {}".format(e), self)
+
+    def _get_csar_version(self):
```

Comment:
```suggestion
    def _validate_csar_version(self):
```

I suggest renaming these methods, because they act more a validator and less a getter.

Reply:
Ok, I agree and this was modified as you suggested.